src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -89,6 +101,10 @@ internal static NamedTypeSymbol Create(SourceModuleSymbol containingModule, stri
             var compilation = containingModule.DeclaringCompilation;
             DiagnosticInfo? diagnosticInfo = null;
 
+            var hasReadOnlyInterfaces =
+                !compilation.IsTypeMissing(SpecialType.System_Collections_Generic_IReadOnlyCollection_T)
+                && !compilation.IsTypeMissing(SpecialType.System_Collections_Generic_IReadOnlyList_T);
```

Comment:
It feels like this should be initialized in terms of the contents of `s_readOnlyInterfacesSpecialTypes`, not repeating those types again here. We should also require the well-known Count and Length members to be present.

I think something like `s_readOnlyInterfacesSpecialTypes.All(...IsPresent) && s_readOnlyInterfacesWellKnownMembers.All(...IsPresent)` would be suitable.

Reply:
It's possible that I missed some reason that existence of the types should be checked here, but not existence of the members. @cston in case you had already given some advice here that I missed, I want to make sure we're in agreement about what to do for this.

Reply:
@RikkiGibson In an earlier form, it would implement the interfaces if present, and then implement each member if the member was present. Does that sound interesting?

Reply:
(Basically, the first commit in the branch)

Reply:
I think that would be fine, although I would suggest checking the members independently, we can simply implement the interfaces and members that happen to exist and don't implement the ones that don't exist.

However, I don't want to lead you around in a circle if Cyrus or Chuck have already steered you away from that path. So let's make sure we have consensus on how that should work.

Reply:
i have zero opinion here.  IT's up to compiler side how you'd like this code to be structured :)